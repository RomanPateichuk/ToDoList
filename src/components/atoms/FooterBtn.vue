<template>
  <label :class="$style.btns_wrapper"
    ><input type="radio" name="radio-footer" @click="Filter" /><span
      :class="$style.radio"
      >{{ FooterBtnValue }}</span
    ></label
  >
</template>

<script>
export default {
  props: {
    FooterBtnValue: {
      type: String,
      value: "",
    },
  },
  methods: {
    Filter: function () {
      this.$emit("callFilter");
    },
  },
};
</script>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
.btns_wrapper {
  input[type="radio"] {
    appearance: none;
  }
  input[type="radio"]:checked {
    + .radio {
      border: 1px solid $footer-border-btn;
      border-radius: 10px;
      box-sizing: border-box;
    }
  }
  .radio {
    border: 1px solid transparent;
    box-sizing: border-box;
    padding: 0.3rem;
  }
}
</style>
