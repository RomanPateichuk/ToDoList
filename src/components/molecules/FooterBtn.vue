<template>
  <label :class="$style.btns_wrapper"
    ><input type="radio" name="radio-footer" /><span
      :class="$style.radio"
      >{{ FooterBtnValue }}</span
    ></label
  >
</template>

<script>
export default {
  props: {
    FooterBtnValue: {
      type: String,
      value: "",
    },
  },
};
</script>

<style lang="scss" module>
@import "@/assets/scss/main.scss";
.btns_wrapper {
  input[type="radio"] {
    appearance: none;
  }

  input[type="radio"]:checked {
    + .radio {
      border: 1px solid $footer-border-btn;
      border-radius: 10px;
      box-sizing: border-box;
    }
  }

  .radio {
    border: 1px solid transparent;
    box-sizing: border-box;
    padding: 0.3rem;
  }
}
</style>
