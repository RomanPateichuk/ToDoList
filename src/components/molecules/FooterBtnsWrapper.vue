<template>
  <div :class="$style.btn">
    <FooterBtn
      v-for="(btn, index) in btns"
      :key="index"
      :FooterBtnValue="btn"
      @callFilter="callMuttationSetFilter(btn)"
    />
  </div>
</template>
<script>
export default {
  components: {
    FooterBtn: () => import("@/components/atoms/FooterBtn.vue"),
  },
  data() {
    return {
      btns: ["All", "Active", "Completed"],
    };
  },
  methods: {
    callMuttationSetFilter: function (FooterBtnValue) {
      this.$store.commit("setFilter", FooterBtnValue);
    },
  },
};
</script>
<style lang="scss" module>
@import "@/assets/scss/main.scss";
.btn {
  :not(:last-child) {
    margin-right: 1rem;
    @include MediaFooterBntmobile;
  }
}
</style>
