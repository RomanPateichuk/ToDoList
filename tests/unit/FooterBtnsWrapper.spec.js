import { mount } from "@vue/test-utils";
import FooterBtnsWrapper from "../../src/components/molecules/FooterBtnsWrapper.vue";
import FooterBtn from "../../src/components/atoms/FooterBtn.vue";

describe("Тестирование компонента FooterProgress:", () => {
  const wrapper = mount(FooterBtnsWrapper, {
    propsData: {},
    mocks: {
      $store: {
        state: {},
        commit: jest.fn(),
      },
    },
  });

  test("соответствие снимку", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Вызов события 'callFilter' из дочернего компонента", () => {
    expect(
      wrapper.findComponent(FooterBtn).vm.$emit("callFilter")
    ).toBeTruthy();
  });

  test("Вызов события 'callRemoveTask' из дочернего компонента", () => {
    expect(
      wrapper.findComponent(FooterBtn).vm.$emit("callFilter")
    ).toBeTruthy();
  });
});
