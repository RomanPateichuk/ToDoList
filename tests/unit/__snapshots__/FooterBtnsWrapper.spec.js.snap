// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента FooterProgress: соответствие снимку 1`] = `<div><label><input type="radio" name="radio-footer"><span>All</span></label><label><input type="radio" name="radio-footer"><span>Active</span></label><label><input type="radio" name="radio-footer"><span>Completed</span></label></div>`;
